google/field_helpers.go

```diff
@@ -220,3 +231,78 @@ func parseOrganizationFieldValue(resourceType, fieldValue string, isEmptyValid b
 
 	return nil, fmt.Errorf("Invalid field format. Got '%s', expected format '%s'", fieldValue, fmt.Sprintf(organizationLinkTemplate, "{org_id}", resourceType, "{name}"))
 }
+
+type RegionalFieldValue struct {
+	Project string
+	Region  string
+	Name    string
+
+	resourceType string
+}
+
+func (f RegionalFieldValue) RelativeLink() string {
+	if len(f.Name) == 0 {
+		return ""
+	}
+
+	return fmt.Sprintf(regionalLinkTemplate, f.Project, f.Region, f.resourceType, f.Name)
+}
+
+// Parses a regional field supporting 5 different formats:
+// - https://www.googleapis.com/compute/ANY_VERSION/projects/{my_project}/regions/{region}/{resource_type}/{resource_name}
+// - projects/{my_project}/regions/{region}/{resource_type}/{resource_name}
+// - regions/{region}/{resource_type}/{resource_name}
+// - resource_name
+// - "" (empty string). RelativeLink() returns empty if isEmptyValid is true.
+//
+// If the project is not specified, it first tries to get the project from the `projectSchemaField` and then fallback on the default project.
+// If the region is not specified, it takes the value of `regionSchemaField`.
```

Comment:
The Regional case is slightly different from the Zonal case. You can set the default region on the provider. The logic should fallback on the default region if it is not specified in the field value or the schema field.

You can refactor the `getRegion` method to have a structure similar to `getProject`. This will introduce a `getRegionFromSchema` method that you can use here.